collectors/nvd/collectors.py

```diff
@@ -324,9 +324,10 @@ def update_cvss_via_flawcvss(self, flaw: Flaw, item: dict) -> bool:
                 if original_cvss != new_cvss:
                     # set impact if it was empty because a validation requires it
                     if not flaw.impact:
-                        flaw.impact = self.calculate_highest_impact(
+                        impact = self.calculate_highest_impact(
                             [new_cvss2_score, new_cvss3_score, new_cvss4_score]
                         )
+                        Flaw.objects.filter(uuid=flaw.uuid).update(impact=impact)
```

Comment:
Can we add a test for this?

Reply:
Could you please check if the tests are enough?
https://github.com/RedHatProductSecurity/osidb/blob/7b7aa58ae0e7ac63b1d88ef0c3cc1fa4bf660141/collectors/nvd/tests/test_collectors.py#L294

Reply:
So the test was already there and did not catch this? :thinking:

Reply:
Yes, do you see any bugs in those tests?

Reply:
Maybe it triggered the `save` in the caller method while the real-life run did not?

Reply:
@jobselko I am not sure. Since the save of the Flaw was only after the CVSS was created/updated, was it really created/updated? Wasn't there some silent fail? That test needs to be definitely investigated as it passed before and it passes now.